src/legacy/plugin_discovery/plugin_spec/plugin_spec_options.d.ts

```diff
@@ -29,4 +29,5 @@ export interface PluginSpecOptions {
   publicDir: string;
   uiExports?: UiExports;
   init: InitPluginFunction;
+  config: any;
```

Comment:
nit: Even though we don't have typings for `Joi`, we could probably at least enforce that config is a function